src/core/annotation.js

```diff
@@ -3932,38 +3960,31 @@ class FreeTextAnnotation extends MarkupAnnotation {
   static createNewDict(annotation, xref, { apRef, ap }) {
     const { color, fontSize, oldAnnotation, rect, rotation, user, value } =
       annotation;
+    const date = `D:${getModificationDate(annotation.date)}`;
```

Comment:
*Here, and elsewhere below:*
Looking at the existing code, it doesn't appear that `annotation.date` is ever being set?

Reply:
I had to add it for the test:
https://github.com/mozilla/pdf.js/pull/19976/files#diff-9bc50ce4ba8a4ba8d322114a3cc05fa62005a40b52223059200ca07087dd5a61R12160
in order to have a fixed reference.

After thinking about that, I think it'd make sense to use it when the annotation has been updated and not when it has been saved. I can fix that in a follow-up.